test-utils/ComponentHarness.js

```diff
@@ -0,0 +1,46 @@
+var Model = require('racer').Model;
+var App = require('../lib/App');
+require('../parsing');
+
+function HarnessApp() {
+  App.call(this);
+}
+HarnessApp.prototype = Object.create(App.prototype);
+// Disable App.prototype._init(), which does setup for loading views
+// from files on the server and loading serialized views and data
+// on the client
+HarnessApp.prototype._init = function() {};
+
+module.exports = ComponentHarness;
+function ComponentHarness() {
+  this.app = new HarnessApp();
+  this.model = new Model();
+  if (arguments.length > 0) {
+    this.setup.apply(this, arguments);
+  }
+}
+ComponentHarness.prototype.setup = function(source) {
+  this.app.views.register('$harness', source);
+  // Remaining variable arguments are components
+  for (var i = 1; i < arguments.length; i++) {
+    var constructor = arguments[i];
+    this.app.component(constructor);
+  }
+  return this;
+};
+ComponentHarness.prototype.getInstance = function() {
+  var page = this._createPage();
+  page.get('$harness');
```

Comment:
Not sure what you're getting at. What alternative were you thinking?

Reply:
I meant that you are doing a .get() but not storing its return value in a variable for later use. If it's necessary then it feels like you may be relying on a side effect of calling .get()

Reply:
Yeah, this is basically a gross hack. `.get()` does HTML rendering, so the return value is HTML. As a side-effect of rendering, components are instantiated, and a reference is kept on `page._components`. Since we just created the page, we can reliably return the first component. It's not pretty, but the other means for getting a reference to a component controller generally rely on adding hooks in the view with `as=` or `on-init=`. Problem with that is we want the user to pass the view in, so they can do whatever setup they want.

It's a hack that leverages an internal API, but it is mitigated by the fact that it is in the Derby repo itself, so if there is a good way of doing this in a more clean manner in the future, we should be able to do that all internally without changing the interface of `getInstance()`.

I should at least document the hack with a comment to explain why.

Reply:
Thanks for taking the time to explain, much appreciated! Yes, a note would be useful for future readers, I think that's a great idea. 👍